lms/static/scripts/frontend_apps/utils/onedrive-picker-client.js

```diff
@@ -71,16 +75,20 @@ export class OneDrivePickerClient {
     return new Promise((resolve, reject) => {
       /** @param {PickerResponse} result */
       const success = result => {
-        // TODO - Clarify nullability of properties and lengths of arrays.
-        // We think there are some situations where these properties won't exist
-        // or the arrays may be empty.
-        //
-        // See https://hypothes-is.slack.com/archives/C2BLQDKHA/p1643309746775569.
-        const driveItem = result.value[0];
-        const name = driveItem.name;
-        const sharingURL = driveItem.permissions[0].link.webUrl;
-        const url = OneDrivePickerClient.downloadURL(sharingURL);
-        resolve({ name, url });
+        try {
+          // @ts-expect-error
+          const driveItem = result.value[0];
```

Comment:
I only capture this very specific error. Other errors are assumed not to be permission errors and are caught and re-thrown on the below try/catch.

Reply:
There is a test to catch these other possible errors.